extensions/ql-vscode/src/remote-queries/run-remote-query.ts

```diff
@@ -298,33 +299,52 @@ export async function prepareRemoteQueryRun(
 }
 
 /**
- * Updates the default suite of the query pack. This is used to ensure
- * only the specified query is run.
+ * Fixes the qlpack.yml file to be correct in the context of the MRVA request.
  *
- * Also, ensure the query pack name is set to the name expected by the server.
+ * Performs the following fixes:
+ *
+ * - Updates the default suite of the query pack. This is used to ensure
+ *   only the specified query is run.
+ * - Ensures the query pack name is set to the name expected by the server.
+ * - Removes any `${workspace}` version references from the qlpack.yml file. Converts them
+ *   to `*` versions.
  *
  * @param queryPackDir The directory containing the query pack
  * @param packRelativePath The relative path to the query pack from the root of the query pack
  */
-async function ensureNameAndSuite(
+async function fixPackFile(
   queryPackDir: string,
   packRelativePath: string,
 ): Promise<void> {
   const packPath = path.join(queryPackDir, "qlpack.yml");
   const qlpack = yaml.load(await fs.readFile(packPath, "utf8")) as QlPack;
-  delete qlpack.defaultSuiteFile;
 
+  // update pack name
   qlpack.name = QUERY_PACK_NAME;
 
-  qlpack.defaultSuite = [
+  // update default suite
+  delete qlpack.defaultSuiteFile;
+  qlpack.defaultSuite = generateDefaultSuite(packRelativePath);
+
+  // remove any ${workspace} version references
+  for (const [key, value] of Object.entries(qlpack.dependencies || {})) {
```

Comment:
Also curious: why the `|| {}`?

Reply:
Just in case `dependencies` is undefined. It's unlikely, but I guess someone _could_ be running a nolang query.

Reply:
Oh, I see, because `Object.entries()` doesn't gracefully handle the case where the input is undefined. Thanks!